node_modules/col-activities/lib/notifications/weekly.js

```diff
@@ -0,0 +1,583 @@
+ * modifications, and distributions.
+ *
+ * Contact The Office of Technology Licensing, UC Berkeley, 2150 Shattuck Avenue,
+ * Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+ * http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+ *
+ * IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+ * INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+ * THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+ * SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+ * "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ * ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+var _ = require('lodash');
+var async = require('async');
+var config = require('config');
+var moment = require('moment-timezone');
+
+var ActivitiesApi = require('col-activities');
+var CollabosphereConstants = require('col-core/lib/constants');
+var CourseAPI = require('col-course');
+var DB = require('col-core/lib/db');
+var EmailUtil = require('col-core/lib/email');
+var log = require('col-core/lib/logger')('col-activities/notifications');
+var UsersAPI = require('col-users');
+var UserConstants = require('col-users/lib/constants');
+
+ /**
+ * Send weekly notification emails for all courses.
+ *
+ * @param  {Function}     callback    Standard callback function
+ */
+var collect = module.exports.collect = function(callback) {
+  var start = Date.now();
+
+  // Get all active courses.
+  CourseAPI.getCourses(function(err, courses) {
+    if (err) {
+      return log.error({'err': err}, 'Unable to retrieve courses for weekly notifications.');
+    }
+
+    // Iterate through all courses.
+    async.eachSeries(courses, collectCourse, function(err) {
+      if (err) {
+        return log.error({'err': err}, 'Unable to collect a course for weekly notifications.');
+      }
+      log.info({'duration': (Date.now() - start)}, 'Sent weekly notifications for all courses.');
+
+      return callback();
+    });
+  });
+};
+
+/**
+ * Send weekly notification emails for a given course.
+ *
+ * Note that this method is only made publicly available so it can be called from the integration
+ * tests. Do not call this method directly.
+ *
+ * @param  {Course}         course          The course for which to send weekly notifications
+ * @param  {Function}       callback        Standard callback function
+ * @api private
+ */
+var collectCourse = module.exports.collectCourse = function(course, callback) {
+  var start = Date.now();
+
+  // 1. Get users and activity data for the course.
+  getCourseData(course, function(err, courseData) {
+    // Don't pass errors up.
```

Comment:
Can you explain why we're not passing them up? Can we at least log an error message when a course has failed?